pyogrio/geopandas.py

```diff
@@ -268,11 +362,13 @@ def read_dataframe(
 
     read_func = read_arrow if use_arrow else read
     gdal_force_2d = False if use_arrow else force_2d
-    if not use_arrow:
-        # For arrow, datetimes are read as is.
-        # For numpy IO, datetimes are read as string values to preserve timezone info
-        # as numpy does not directly support timezones.
-        kwargs["datetime_as_string"] = True
+
+    # Always read datetimes as string values to preserve (mixed) timezone info
+    # correctly. If arrow is not used, it is needed because numpy does not
+    # directly support timezones. If arrow is used, needed because datetime
+    # columns don't support mixed timezone offsets + e.g. for .fgb files
+    # timezone info isn't handled correctly even for unique timezone offsets
+    # if datetimes are not read as string.
```

Comment:
This is what was discussed in https://github.com/OSGeo/gdal/issues/11212, i.e. it is not considered a bug on the GDAL side, but just a limitation of the combo of fgb/gpkg (which can have mixed offsets, information not known at the column level) and arrow (needs tz information at the column level, and we don't know the tz before parsing the full file, thus always normalize to UTC)?

Reply:
The problem described here was solved in `ogr2ogr` the same way as in `pyogrio`: by reading datetimes as string via arrow.

The PR that fixes this in `ogr2ogr` is referenced by rouault in this comment: https://github.com/OSGeo/gdal/issues/11212#issuecomment-2462091204